core/logic_events.go

```diff
@@ -695,6 +725,64 @@ func (e eventsLogic) formatDate(wtDate string) string {
 	return result
 }
 
+func (e eventsLogic) preProcessLocation(allWebtoolsEvents []model.WebToolsEvent) error {
+	locationEventMap := make(map[string]map[string]string)
+
+	for _, event := range allWebtoolsEvents {
+		locationEventMap[event.EventID] = map[string]string{
+			"eventID":      event.EventID,
+			"location":     event.Location,
+			"sponspor":     event.Sponsor,
+			"calendarName": event.CalendarName,
+		}
+	}
+
+	locationFromTheDatabase, err := e.app.storage.FindLegacyLocationItems()
+	if err != nil {
+		return nil
+	}
+
+	newLocations := make(map[string]map[string]string)
+
+	for eventID, eventLocation := range locationEventMap {
+		found := false
+		for _, dbLocation := range locationFromTheDatabase {
+			if eventLocation["location"] == dbLocation.Name {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			newLocations[eventID] = map[string]string{
+				"ID":      eventID,
+				"Name":    eventLocation["location"],
+				"Sponsor": eventLocation["sponsor"],
+			}
+		}
+	}
+
+	// Process the values in the map
+	for eventID, eventData := range newLocations {
+		l, _ := e.geoBBAdapter.ProcessLocation(eventID, eventData["Name"], eventData["Sponsor"], eventData["Name"])
+		err = e.app.storage.InsertLegacyLocationItem(*l)
+		if err != nil {
+			return nil
+		}
+
+	}
+
+	return nil
+}
+
+func (e eventsLogic) processLocation() ([]model.LegacyLocation, error) {
```

Comment:
This function name is misleading. It does not process location. Please put this into the preprocessing function as it has been done for the images.